File: backend/app/schemas/catalog.py
from pydantic import BaseModel, ConfigDict


class ProductCategoryOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    sort_order: int


class ProductOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    restaurant_id: int
    category_id: int
    name: str
    description: str | None
    price: float
    image_url: str | None
    is_active: bool
    sort_order: int
    category: ProductCategoryOut


class ProductListOut(BaseModel):
    items: list[ProductOut]


class ProductUpsertResult(BaseModel):
    item: ProductOut


class ProductCategoryListOut(BaseModel):
    items: list[ProductCategoryOut]


class ProductCategoryCreateRequest(BaseModel):
    name: str
    sort_order: int | None = None


class ProductCategoryCreateResult(BaseModel):
    item: ProductCategoryOut


class ProductCategoryUpdateRequest(BaseModel):
    name: str
    sort_order: int = 0

File: frontend/src/types/catalog.ts
export type Category = {
  id: number
  name: string
  sort_order: number
}

export type CategoryFormState = {
  name: string
  sortOrder: string
}

export type Product = {
  id: number
  restaurant_id: number
  category_id: number
  name: string
  description: string | null
  price: number
  image_url: string | null
  is_active: boolean
  sort_order: number
  category: Category
}

export type ProductFormState = {
  name: string
  categoryId: string
  price: string
  description: string
  sortOrder: string
  isActive: boolean
}

export const emptyProductForm = (): ProductFormState => ({
  name: '',
  categoryId: '',
  price: '',
  description: '',
  sortOrder: '0',
  isActive: true
})

export const emptyCategoryForm = (): CategoryFormState => ({
  name: '',
  sortOrder: '0'
})
